pilot/pkg/proxy/envoy/v2/debug.go

```diff
@@ -929,3 +901,16 @@ func printRoutes(w io.Writer, c *XdsConnection) {
 		}
 	}
 }
+
+func (s *DiscoveryServer) getProxyConnection(proxyID string) *XdsConnection {
+	s.adsClientsMutex.RLock()
+	defer s.adsClientsMutex.RUnlock()
+
+	for conID := range s.adsClients {
+		if strings.Contains(conID, proxyID) {
```

Comment:
Is there any specific reason why we can not have adsClients just store the proxyID instead of appending with some number? So that this just becomes a map lookup?

Reply:
I think you can have multiple connections from a single proxy. Not very common but at least during hot restart it could? Or if the connection isn't properly closed or whatever other reason

Reply:
yep

Reply:
Makes sense. Hot restart completely slipped out of my mind :-(